frontend/src/main/scala/bloop/bsp/BspServer.scala

```diff
@@ -0,0 +1,110 @@
+package bloop.bsp
+
+import java.net.ServerSocket
+import java.util.Locale
+import java.nio.file.{Files, Path}
+
+import bloop.cli.{BspProtocol, Commands}
+import bloop.engine.{ExecutionContext, State}
+import com.martiansoftware.nailgun.{NGUnixDomainServerSocket, NGWin32NamedPipeServerSocket}
+import monix.execution.{Cancelable, Scheduler}
+
+object BspServer {
+  private[bloop] val isWindows: Boolean =
+    System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("windows")
+  private[bloop] val isMac: Boolean =
+    System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("mac")
+
+  import java.net.InetSocketAddress
+  private sealed trait ConnectionHandle { def serverSocket: ServerSocket }
+  private case class WindowsLocal(pipeName: String, serverSocket: ServerSocket)
+      extends ConnectionHandle
+  private case class UnixLocal(path: Path, serverSocket: ServerSocket) extends ConnectionHandle
+  private case class Tcp(address: InetSocketAddress, serverSocket: ServerSocket)
```

Comment:
I thought that we agreed that `address` was not necessary? Is it in the end?

Reply:
`address` is necessary because otherwise it's not possible to know the port from `serverSocket.getAddress`, since it returns an `InetAddress`, its super class.